tests/unit/test_merlion_handler.py

```diff
@@ -0,0 +1,224 @@
+        test["train"] = 0
+        df = pd.concat([train, test], axis=0)
+        return df
+
+    def get_nab_df(self) -> pd.DataFrame:
+        df = pd.read_csv("https://raw.githubusercontent.com/numenta/NAB/master/data/realKnownCause/nyc_taxi.csv")
+        df.rename(columns={"timestamp": "t", "value": "val"}, inplace=True)
+        train_len = int(len(df) * 0.5)
+        df_train = df.iloc[: train_len]
+        df_test = df.iloc[train_len: ]
+        df_train["train"] = 1
+        df_test["train"] = 0
+        df = pd.concat([df_train, df_test], axis=0)
+        return df
+
+    def run_mindsdb_sql(self, sql):
+        return self.command_executor.execute_command(
+            parse_sql(sql, dialect='mindsdb')
+        )
+
+    def test_merlion_forecaster(self):
+        df = self.get_m4_df()
+        df_train = df[df["train"] == 1][["H1"]]
+        df_test = df[df["train"] == 0][["H1"]]
+        # default adapter
+        adapter = DefaultForecasterAdapter()
+        adapter.train(df_train.copy(deep=True), target="H1")
+        rt_df = adapter.predict(df_test.copy(deep=True), target="H1")
+        assert rt_df is not None
+        assert len(rt_df) == len(df_test)
+        assert "H1__upper" in set(rt_df.columns.values) and "H1__lower" in set(rt_df.columns.values)
+        # arima adapter
+        adapter = SarimaForecasterAdapter()
+        adapter.train(df_train.copy(deep=True), target="H1")
+        rt_df = adapter.predict(df_test.copy(deep=True), target="H1")
+        assert rt_df is not None
+        assert len(rt_df) == len(df_test)
+        assert "H1__upper" in set(rt_df.columns.values) and "H1__lower" in set(rt_df.columns.values)
+        # prophet adapter
+        adapter = ProphetForecasterAdapter()
+        adapter.train(df_train.copy(deep=True), target="H1")
+        rt_df = adapter.predict(df_test.copy(deep=True), target="H1")
+        assert rt_df is not None
+        assert len(rt_df) == len(df_test)
+        assert "H1__upper" in set(rt_df.columns.values) and "H1__lower" in set(rt_df.columns.values)
+        # mses adapter
+        adapter = MSESForecasterAdapter()
+        adapter.train(df_train.copy(deep=True), target="H1")
+        rt_df = adapter.predict(df_test.copy(deep=True), target="H1")
+        assert rt_df is not None
+        assert len(rt_df) == len(df_test)
+        assert "H1__upper" in set(rt_df.columns.values) and "H1__lower" in set(rt_df.columns.values)
+
+    @patch('mindsdb.integrations.handlers.postgres_handler.Handler')
+    def test_merlion_forecaster_sql(self, mock_handler):
+        # prepare data
+        df = self.get_m4_df()
+        df["t"] = df.index
+        self.set_handler(mock_handler, name='pg', tables={'m4': df})
+        # test default
+        self.exec_train_and_forecast(mock_handler=mock_handler, model_name="default", using="")
+        # # test arima
+        # self.exec_train_and_predict(mock_handler=mock_handler, model_name="arima", using=f"USING model_type='arima'")
```

Comment:
Curious as to why these are disabled? Presumably it's fast enough when fitting a single series, correct?

Reply:
ok, I will uncomment it.

Reply:
I was just curious 😛 . Enable the ones you deem necessary for testing this handler thoroughly, and the rest can be removed (not necessary to have them as comments).